block/block.go

```diff
@@ -1,13 +1,42 @@
 package block
 
 import (
+	"context"
+	"errors"
 	"fmt"
 
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
+
 	errorsmod "cosmossdk.io/errors"
 
 	"github.com/dymensionxyz/dymint/types"
 )
 
+// applyBlockWithFraudHandling calls applyBlock and validateBlockBeforeApply with fraud handling.
+func (m *Manager) applyBlockWithFraudHandling(block *types.Block, commit *types.Commit, blockMetaData types.BlockMetaData, validate bool) error {
+	if validate {
+		if err := m.validateBlockBeforeApply(block, commit); err != nil {
+			if errors.Is(err, gerrc.ErrFault) {
+				m.FraudHandler.HandleFault(context.Background(), err)
+			} else if err != nil {
```

Comment:
not sure if there exists the case where it returns an error that is actually not a fraud.